hkmc2/shared/src/main/scala/hkmc2/semantics/Symbol.scala

```diff
@@ -36,13 +36,19 @@ abstract class Symbol(using State) extends Located:
     case mem: BlockMemberSymbol =>
       mem.modTree.map(_.symbol.asInstanceOf[ModuleSymbol])
     case _ => N
+  def asPat: Opt[PatternSymbol] = this match
+    case pat: PatternSymbol => S(pat)
+    case mem: BlockMemberSymbol =>
+      mem.patTree.map(_.symbol.asInstanceOf[PatternSymbol])
```

Comment:
This should probably be
```suggestion
      mem.patTree.flatMap(_.symbol.asPat)
```
(create the `asPat` method if it's not already there)

Reply:
Thanks. I guess I need to modify other methods as well?

https://github.com/chengluyu/mlscript/blob/37e9025c1af976d79b97a40766f8f5a37cef76bd/hkmc2/shared/src/main/scala/hkmc2/semantics/Symbol.scala#L29-L48

Reply:
Oh, true. Sure.